### PR TITLE
opendatahub-io/caikit-nlp: add release workflows

### PR DESCRIPTION
**Pull request by dtrifiro**
- opendatahub-io/caikit-nlp: add workflow for release branch
- opendatahub-io/caikit-nlp: add workflow for red-hat-data-services

### CODE REVIEW

**Comment by dtrifiro** [comment]
@Xaenalt

**Review by Xaenalt — APPROVED** [review]

**Comment by dtrifiro** [comment]
regarding the above failure: I ended up removing the red hat data services jobs

File: ci-operator/config/opendatahub-io/caikit-nlp/opendatahub-io-caikit-nlp-release.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  to: caikit-nlp
+promotion:
+  namespace: opendatahub-io
+  tag: stable
+releases:
+  latest:
+    release:
+      architecture: amd64
+      channel: fast
+      version: "4.12"
```
Comment (dtrifiro) [inline ci-operator/config/opendatahub-io/caikit-nlp/opendatahub-io-caikit-nlp-release.yaml:18]: @Xaenalt @heyselbi @rpancham  do you know if setting `version: "4.12"` is alright here?

**Pull request merged**